File: src/common_letters.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

string common_letters(const vector<string>& words) {
    if (words.empty()) {
        return "";
    }

    string common = words[0];
    for (int i = 1; i < words.size(); ++i) {
        string word = words[i];
        sort(common.begin(), common.end());
        sort(word.begin(), word.end());
        string temp;
        set_intersection(common.begin(), common.end(), word.begin(), word.end(), back_inserter(temp));
        common = temp;
    }

    return common;
}

int main() {
    vector<string> words;
    string word;
    while (cin >> word) {
        words.push_back(word);
    }

    string result = common_letters(words);
    cout << result << endl;

    return 0;
}

File: src/common_letters.rs
use std::collections::HashSet;
use std::io::{self, BufRead};

fn common_letters(words: Vec<String>) -> String {
    if words.is_empty() {
        return String::new();
    }

    let mut common: HashSet<char> = words[0].chars().collect();
    for word in words.iter().skip(1) {
        let word_set: HashSet<char> = word.chars().collect();
        common = common.intersection(&word_set).cloned().collect();
    }

    let mut result: Vec<char> = common.into_iter().collect();
    result.sort();
    result.into_iter().collect()
}

fn main() {
    let stdin = io::stdin();
    let words: Vec<String> = stdin.lock().lines().filter_map(|line| line.ok()).collect();
    
    let result = common_letters(words);
    println!("{}", result);
}